src/routing/lazy_messaging.rs

```diff
@@ -36,58 +35,65 @@ pub(crate) fn process(
         return Ok(actions);
     }
 
-    if let Ok(src_key) = msg.proof_chain_last_key() {
-        if !network.has_key(src_key) && !section.chain().has_key(src_key) {
-            // We don't know the src key. Send them a `OtherSection` message whose `dst_key` is set
-            // to the latest key we know and when they receive it they would send us back a
-            // `OtherSection` with their latest info, including their latest key.
-            //
-            // NOTE: the reason why we update the key only after receiving their `OtherSection` and
-            // not right here is to keep their key and their elders info in sync.
-            trace!("lazy messaging: src key unknown");
-            send_other_section = true;
+    if let Ok(src_chain) = msg.proof_chain() {
+        if let Some(key) = network.key_by_name(&src_name) {
+            match src_chain.cmp_by_position(src_chain.last_key(), key) {
+                Ordering::Greater => {
+                    trace!("Anti-Entropy: We do not know source's key, need to update ourselves");
```

Comment:
>send an OtherSection to the sender to notify it that we now get the knowledge updated to the newer key

Are you sure? We don't need to tell anyone we updated? We may resend a message that required us to be updated but we don't tell folk when we update.

Reply:
>notify it that we now get the knowledge updated to the newer key

This should be with the `knowledge` map we are using. Since we'll be removing that in this PR, we don't need to notify other sections now and keep it just as individual messages.